src/rust/catmem/mod.rs

```diff
@@ -153,7 +136,7 @@ impl CatmemLibOS {
                 // Release the queue descriptor, even if pushing EoF failed. This will prevent any further
                 // operations on the queue, as well as it will ensure that the underlying shared ring buffer will
                 // be eventually released.
-                qtable.borrow_mut().free(&qd);
+                runtime.free_queue::<CatmemQueue>(&qd).expect("queue should exist");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?